include/deal.II/lac/vector_memory.templates.h

```diff
@@ -42,18 +43,12 @@ inline
 GrowingVectorMemory<VectorType>::Pool::~Pool()
 {
   // Nothing to do if memory was unused.
-  if (data == nullptr) return;
-
-  // First, delete all remaining
-  // vectors. Actually, there should
-  // be none, if there is no memory
-  // leak
-  for (typename std::vector<entry_type>::iterator i=data->begin();
-       i != data->end();
-       ++i)
-    {
-      delete i->second;
-    }
+  if (data == nullptr)
+    return;
+
+  // delete the 'data' object. this also releases all vectors
+  // that are pointed to by the std::unique_ptrs
+  data->clear();
   delete data;
```

Comment:
Actually, I take that back. This code is in the destructor, so the memory is released. 

But I still don't know the answer to your question. I decided to do one thing at a time :-)